### DISCUSSION TITLE
Compiling the project

### DISCUSSION THREAD

**Discussion by danieledellacioppa** [mapsforge/mapsforge#1304] [Q&A] [answered]
Hi guys,

I'm on an apprenticeship and I've been asked to make an app to overlay some objects on a map.

I'm completely new to mobile development
Nobody can help me at the college. It's all on me and google but I'm tired of googling and want to see if someone can point me in the right direction once for all

What I did is:
cloning the repo and opening it as a project (since I've seen some gradle.settings in it I assumed it would work as a project)
I synced. It gave me this error 

> /home/daniele/Documents/programming/nongit/mapsforge/mapsforge-map-android/src/main/java/org/mapsforge/map/android/graphics/AndroidPointTextContainer.java:18: error: package android.text does not exist
> import android.text.Layout;
>                    ^

I added android.enableJetifier=true in the gradle.properties since it helped me to build maptiler (another openstreetmaps library close to sunset) but the error code still doesn't change.

If I take a single folder that Android Studio is showing me with the same icon as a project and then try to sync and build, no matter what's the project, where it comes from, it always complains about a missing "wrapper" so I guess I'm not doing it right

I'm on Linux Mint, using Android Studio Bumblebee 2021.1.1 

What I need is just to import a plugin and overlay some (x,y) points on a map. I've tried these steps with maptiler and because the project is not as complex as this one it all works but maptiler is close to sunset and I'm helpless so I'm asking it here

If somehow my information is incomplete ask me I'll answer promptly. That's the job I'm supposed to be doing anyway. I'll be prompt

Thanks

Daniele

**Comment by devemux86** [comment]
To compile the map library see here: https://github.com/mapsforge/mapsforge/discussions/1291#discussioncomment-2147373

---

It is simpler to use the map library dependencies,
see the instructions in the [integration guide](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md).

And study the various samples, e.g. [OverlayMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java).

**Comment by danieledellacioppa** [comment]
Hi I'm writing step by step every line of code to make sure if I get an error it'll be just one.

[This is the code I'm trying to study ](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/GettingStarted.java#L30) but the last library cannot be resolved.

it's called InternalRenderTheme.

I just want to make the easiest project ever just showing a map I can zoom in and make on it other gestures with my fingers ( I hope this is actually simple )

Can I replace this library with something else? Is it a PATH issue? 

Here goes the output I get:


```
import org.mapsforge.map.rendertheme.InternalRenderTheme;
                                     ^
  symbol:   class InternalRenderTheme
  location: package org.mapsforge.map.rendertheme
```

essentially my understand is that rendertheme hasn't got the InternalRenderTheme anymore and something else needs to be used instead but this is just a guess.

Thanks

Daniele

**Comment by devemux86** [comment]
You need to include the library dependencies in your app.
See the [integration guide](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md) for the details.

**Comment by danieledellacioppa** [comment]
Do you mean adding this lines in the build.gradle file?

    //MAP
    implementation 'org.mapsforge:mapsforge-map-android:0.17.0'
    implementation 'com.caverock:androidsvg:1.4'

    //POI
    implementation 'org.mapsforge:mapsforge-poi-android:0.17.0'
    implementation 'org.mapsforge:sqlite-android:0.17.0'
    implementation 'org.mapsforge:sqlite-android:0.17.0:natives-armeabi-v7a'
    implementation 'org.mapsforge:sqlite-android:0.17.0:natives-arm64-v8a'
    implementation 'org.mapsforge:sqlite-android:0.17.0:natives-x86'
    implementation 'org.mapsforge:sqlite-android:0.17.0:natives-x86_64'
    
    That's my way of adding libraries as far I am aware
Am I doing this wrong?

**Comment by devemux86** [comment, marked as answer]
There are different sections in the integration guide depending on which components you want to use.

For the map on Android you need:
- https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md#core
- https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md#android

**Comment by danieledellacioppa** [comment]
Thanks very much for your prompt answer

It was just me missing core libraries

That problem is solved. It syncs, builds and runs

Thanks

Daniele
